src/test/java/io/wgj/ctci/ArrayListTest.java

```diff
@@ -4,132 +4,108 @@
 import org.junit.Test;
 import org.junit.rules.ExpectedException;
 
-import java.util.Iterator;
-
 import static org.junit.Assert.*;
 
 public class ArrayListTest {
   @Rule
   public final ExpectedException exception = ExpectedException.none();
 
   @Test
-  public void arrayList_new() {
-    ArrayList<Integer> al = new ArrayList<>();
-  }
-
-  @Test
-  public void arrayList_add() {
-    ArrayList<Integer> al = new ArrayList<>();
-    al.add(1);
-  }
-
-  @Test
-  public void arrayList_size() {
-    ArrayList<Integer> al = new ArrayList<>();
-    assertEquals(0, al.size());
-    al.add(1);
-    assertEquals(1, al.size());
-  }
-
-  @Test
-  public void arrayList_grow() {
-    ArrayList<Integer> al = new ArrayList<>();
-    assertEquals(0, al.size());
+  public void arrayList() {
+    ArrayList<Integer> list = new ArrayList<>();
+    assertEquals(0, list.size());
     Integer[] ints = {1, 2, 3, 4, 5};
     for (Integer i : ints) {
-      al.add(i);
+      list.add(i);
     }
-    assertEquals(5, al.size());
-  }
-
-  @Test
-  public void arrayList_get() {
-    ArrayList<Integer> al = new ArrayList<>();
-    al.add(1);
-    assertEquals((Integer) 1, al.get(0));
+    assertEquals(5, list.size());
+    assertEquals((Integer) 1, list.get(0));
+    assertEquals((Integer) 5, list.get(list.size()-1));
   }
 
   @Test
   public void arrayList_rangeCheck() {
-    ArrayList<Integer> al = new ArrayList<>();
+    ArrayList<Integer> list = new ArrayList<>();
     exception.expect(IndexOutOfBoundsException.class);
-    al.get(0);
+    list.get(0);
   }
 
+  private boolean compareArrayToArrayList(Integer[] array, ArrayList<Integer> list) {
+    if (array.length != list.size()) {
+      return false;
```

Comment:
IMHO it would be cleaner to have this method do the asserts directly, ie assertEquals(array.length, list.size())
That would make tracking down failures easier too, since juint would show which line ti failed on.